Extension/src/LanguageServer/client.ts

```diff
@@ -2437,6 +2439,15 @@ export class DefaultClient implements Client {
         });
     }
 
+    /**
+     * send notifications to the language server to restart IntelliSense for the selected file.
+     */
+    public async restartIntelliSenseForFile(document: vscode.TextDocument): Promise<void> {
+        this.notifyWhenLanguageClientReady(() => {
```

Comment:
This can be simplified to 
await this.awaitUntilLanguageClientReady();

A few other spots could also be simplified.

Reply:
Sean, I think this is fine, I know we talked about it. But I can see in the code that wherever there is a notification, we have called `notifyWhenLanguageClientReady`, and we called `awaitUntilLanguageClientReady` otherwise.

```
    public notifyWhenLanguageClientReady<T>(notify: () => T): Promise<T> {
        const task: () => Promise<T> = () => new Promise<T>(resolve => {
            resolve(notify());
        });
        return this.queueTask(task);
    }

    public awaitUntilLanguageClientReady(): Thenable<void> {
        const task: () => Thenable<void> = () => new Promise<void>(resolve => {
            resolve();
        });
        return this.queueTask(task);
    }
```

Reply:
Calling the notify* isn't necessary in async functions. It makes the code simpler to call `await this.await*`. If other async methods are also calling the notify* they could also be changed (I already changed some of those in my clangTidy branch).

Reply:
Well, notify could be okay too if you wanted to make it all on one line and remove the 2 extra lines, i.e. `() => this.languageClient.sendNotification(RestartIntelliSenseForFileNotification, this.languageClient.code2ProtocolConverter.asTextDocumentIdentifier(document)); `.